gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -364,76 +426,79 @@ public void checkAccess(Path path, AccessMode... modes) throws IOException {
     if (cloudPath.seemsLikeADirectoryAndUsePseudoDirectories()) {
       return;
     }
-    if (gcsService.getMetadata(cloudPath.getGcsFilename()) == null) {
+    if (gcsStorage.get(cloudPath.getBlobId()) == null) {
       throw new NoSuchFileException(path.toString());
     }
   }
 
   @Override
-  public <A extends BasicFileAttributes> A readAttributes(
-      Path path, Class<A> type, LinkOption... options) throws IOException {
-    CloudStoragePath cloudPath = checkPath(path);
-    checkNotNull(type);
-    checkNotNullArray(options);
-    if (type != CloudStorageFileAttributes.class && type != BasicFileAttributes.class) {
-      throw new UnsupportedOperationException(type.getSimpleName());
-    }
-    if (cloudPath.seemsLikeADirectoryAndUsePseudoDirectories()) {
-      @SuppressWarnings("unchecked")
-      A result = (A) CloudStoragePseudoDirectoryAttributes.SINGLETON_INSTANCE;
-      return result;
-    }
-    GcsFileMetadata metadata = gcsService.getMetadata(cloudPath.getGcsFilename());
-    if (metadata == null) {
-      throw new NoSuchFileException(path.toString());
-    }
-    @SuppressWarnings("unchecked")
-    A result = (A) new CloudStorageObjectAttributes(metadata);
-    return result;
-  }
-
-  @Override
   public <V extends FileAttributeView> V getFileAttributeView(
       Path path, Class<V> type, LinkOption... options) {
-    CloudStoragePath cloudPath = checkPath(path);
     checkNotNull(type);
     checkNotNullArray(options);
     if (type != CloudStorageFileAttributeView.class && type != BasicFileAttributeView.class) {
       throw new UnsupportedOperationException(type.getSimpleName());
     }
+    CloudStoragePath cloudPath = checkPath(path);
     @SuppressWarnings("unchecked")
-    V result = (V) new CloudStorageFileAttributeView(this, cloudPath);
+    V result = (V) new CloudStorageFileAttributeView(gcsStorage, cloudPath);
     return result;
   }
 
-  /** Does nothing since GCS uses fake directories. */
+  /** Does nothing since GCS uses fake directories.
+   */
   @Override
   public void createDirectory(Path dir, FileAttribute<?>... attrs) {
     checkPath(dir);
     checkNotNullArray(attrs);
   }
 
-  /** @throws UnsupportedOperationException */
+  /** Always @throws UnsupportedOperationException.
+   */
   @Override
   public DirectoryStream<Path> newDirectoryStream(Path dir, Filter<? super Path> filter) {
     // TODO(b/18997618): Implement me.
     throw new UnsupportedOperationException();
   }
 
-  /**
-   * This feature is not supported. Please use {@link #readAttributes(Path, Class, LinkOption...)}
-   *
-   * @throws UnsupportedOperationException
-   */
+  @Override
+  public <A extends BasicFileAttributes> A readAttributes(
```

Comment:
I was getting a linter error because it needs the two methods to be next to each other. I moved it up, but will probably have to move it back down later.